samza-azure/src/main/java/org/apache/samza/system/eventhub/producer/EventHubSystemProducer.java

```diff
@@ -245,6 +254,10 @@ public synchronized void flush(String source) {
     if (!isStarted) {
       throw new SamzaException("Trying to call send before the producer is started.");
     }
+    if (!isInitialized) {
+      // lazy initialization on the first send
+      init();
```

Comment:
@lhaiesp thanks for the workaround. it will also be good to understand why the `PartitionSender` does not shut-down cleanly - maybe there's a bug in the client or we use the client is incorrectly? Can you create a ticket for it and follow-up with the EventHubs team (if need be)?

Reply:
Created the ticket:

https://issues.apache.org/jira/browse/SAMZA-1708